new_csaf/csaf/test/base.py

```diff
@@ -1,4 +1,234 @@
-class CsafTest:
-    pass
+from os import stat
+import csaf
+from f16lib.acas import AcasScenario, AcasShieldScenario, AcasScenarioViewer
 
+from collections.abc import Callable
 
+from abc import ABC, abstractmethod
+
+import numpy as np
+import typing
+import GPyOpt
+import GPy
+
+class Optimizer(ABC):
+    """
+    Generic optimizer
+    """
+    @abstractmethod
+    def prepare(self,
+                configuration_space: dict,
+                obj_func: Callable[[[float]], float]):
```

Comment:
Typing would be `Callable[Sequence[Any], Any]` or `Callable[Sequence[Any], float]`. There's not a lot we can impose of what is being optimized.